src/Servers/Kestrel/Transport.Libuv/src/Internal/LibuvThread.cs

```diff
@@ -76,6 +75,8 @@ public LibuvThread(LibuvFunctions libuv, IHostApplicationLifetime appLifetime, M
 
         public WriteReqPool WriteReqPool { get; }
 
+        public Action<Exception> ShutdownCallback { get; set; }
```

Comment:
Make this a ThreadLifetimeTask instead.

```suggestion
        public Task ThreadLifetimeTask { get; }
```

Reply:
I've made the change. Please have a look.